### PR TITLE
refactor: useBlockSettings to use custom event instead of window

### PR DESCRIPTION
**Pull request by SamuelAlev**
To test:
Link App Bridge on clarify and in a block
- app-bridge: `git checkout origin/refactor/change-useblocksettings && pnpm build`
- clarify: `pnpm link ../app-bridge`
- one guideline-block: `pnpm link ../../../app-bridge`

### CODE REVIEW

File: src/react/useBlockSettings.ts [hunk 1]
```diff
@@ -10,30 +13,43 @@ export const useBlockSettings = <T = Record<string, unknown>>(
...
+        };
+
+        if (blockId) {
+            const mountingFetch = async () => {
+                const allBlockSettings = await appBridge.getBlockSettings<T>();
+                setBlockSettings(allBlockSettings);
+            };
+            mountingFetch();
+
+            window.emitter.on('StyleguideBlockSettingsUpdated', updateBlockSettingsFromEvent);
```
Comment (triggertoo) [inline src/react/useBlockSettings.ts:34]: We could namespace the event, so we know the source of the event, i.e. `app-bridge:updateBlockSettingsFromEvent `. wdyt?

**Comment by triggertoo** [comment]
i just linked it and changed some settings => looks good from my point of view! 👍

File: src/react/useBlockSettings.ts [hunk 2]
```diff
@@ -10,30 +13,43 @@ export const useBlockSettings = <T = Record<string, unknown>>(
...
-                return true;
-            },
-        });
-    }
+    const [blockSettings, setBlockSettings] = useState<T>({} as T);
+
+    // Fetch the block settings on mount.
+    // And add listener for block settings updates.
+    useEffect(() => {
+        const updateBlockSettingsFromEvent = (event: { blockId: number; blockSettings: unknown }) => {
```
[the remark this answers is not in the document]
Comment (SamuelAlev) [reply src/react/useBlockSettings.ts]: Added a generic type

File: src/react/useBlockSettings.ts (hunk 1 above)
Comment (SamuelAlev) [reply src/react/useBlockSettings.ts]: Done

**Review by triggertoo — APPROVED** [review]

**Pull request merged**